### DISCUSSION TITLE
Floating Window Border Color

### DISCUSSION THREAD

**Discussion by yngwi** [nvim-neo-tree/neo-tree.nvim#35] [Q&A] [answered]
Hi,
thank you for your interesting plugin. Is it / will it be possible to change the color of the border around the floating window? I would like to have it look the same as my other floating windows (Telescope, for instance).
Thanks,
Daniel

**Comment by cseickel** [comment]
Absolutely! There are two settings to control the borders. First, in your call to setup you can change the style. The default is "NC", but you may want to change it to "rounded" to match Telescope's default:

```lua
require("neo-tree").setup({
  popup_border_style = "rounded"
})
```

Second, the highlight group used is called NeoTreeFloatBorder, which is linked to FloatBorder by default. If you set either one of those highlight groups, it will change the color of your borders. You could just link it to Telescope's highlight group somewhere in your vim config:

```lua
vim.cmd([[highlight link NeoTreeFloatBorder TelescopeBorder]])

require("neo-tree").setup({
  popup_border_style = "rounded"
})
```

**Reply by yngwi** [reply]
Thank you that works. Could you also tell me the highlight group for the window title "Neo-tree Filesystem"? Also, is it possible to set "q" as a key to close the window if it is open?

I'm also having an error when toggling the _other_ view, so if I first open the floating window and then try to open the sidebar buffer version and vice versa:
![image](https://user-images.githubusercontent.com/963440/148773538-5c642b19-0354-42cb-b986-5b59b095a914.png)

**Comment by cseickel** [comment]
The title bar highlight group is called `NeoTreeTitleBar`.

You can add a custom command and mapping to close it with q, I'll post that example later.

The bug you are hitting was just fixed this morning, it should go away if you update.

**Comment by cseickel** [comment]
Here's how to add a custom close command:
```lua
      require("neo-tree").setup({
        filesystem = {
          window = {
            mappings = {
              ["q"] = "close_window"
            },
          },
          commands = {
            close_window = function(state)
              require("neo-tree").close(state.name)
            end,
        },
      })
```

**Comment by yngwi** [comment]
Hi again, thank you for your help everything works except the title text color, even after setting this:

```
vim.cmd([[highlight! link NeoTreeFloatBorder TelescopeBorder]])
vim.cmd([[highlight! link NeoTreeTitleBar TelescopeBorder]])
```

The text looks like this: 
![image](https://user-images.githubusercontent.com/963440/148836140-8bb3da41-0ee8-448d-9e8e-3432b0f2be34.png)

I would be grateful for pointers fixing this.

**Comment by cseickel** [comment, marked as answer]
What I can tell you is that the NeoTreeTitleBar group is only used when the border style is set to "NC". I'm not exactly sure what the title text will be otherwise, but I can tell you I am using the NuiPopup to render this: https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/popup

My guess is that the text is just defaulting to Normal (or NormalNC, if you use that). You might want to just set the NeoTreeFloatBorder background to match your Normal Background to make everything blend in, or just set that background to NONE to clear it.

**Reply by yngwi** [reply]
Thank you. I have been trying to look it it up and have made some progress but in the end I couldn't figure it out so I'll just leave it white for now. It doesn't matter too much :)

**Reply by hbiel** [reply]
Hi everyone!

I just ran into this too and fixed it by using the NuiText-Object in the NeoTreeTitleBar group in both cases, not just for the NC border style variant. See this commit: https://github.com/hbiel/neo-tree.nvim/commit/fc695fb0e017b9fc4e211d6c51f292869a5eed5b

@cseickel Is this be something you'd accept as a pull request?

**Reply by cseickel** [reply]
> @cseickel Is this be something you'd accept as a pull request?

Absolutely!
